### ISSUE TITLE
Installing from flathub always give the error: Missing ROM file!

### ISSUE THREAD

**Issue by gabaozin** [mike632t/x11-calc#38]
Installing from flathub always gives the error: Missing ROM file!

**Comment by mike632t** [comment]
Not always - only the Voyager series models (HP10C, HP11C, HP12C, HP15C, and HP16C) will be affected.

Unfortunately I don't currently believe that I'm able to include the necessary ROMs for those models.  This means that unlike all the other emulators the ROM images for the HP10C, HP11C, HP12C, HP15C, HP16C are not defined in the the source code and must be loaded from a separate ROM file.

However, there are other sources for the missing ROMs the easiest of which can currently be found [here](https://github.com/brouhaha/nonpareil/tree/main/ncd).  Look for the ROM files in the folders for the Voyager models.

When installed using Flatpak when loading one of the Voyager emulators the application launcher will automatically search for a ROM file that corresponds to the current model in the following location :
```
$HOME/.var/app/io.github.mike632t.x11-calc/data/x11-calc/
``` 
So using the HP11C as an example you can download the appropriate ROM and move it to the correct folder using the following commands (Note the destination filename):
```
wget https://raw.githubusercontent.com/brouhaha/nonpareil/main/ncd/11c/11c.rom
mv 11c.rom $HOME/.var/app/io.github.mike632t.x11-calc/data/x11-calc/x11-calc-11c.rom
```
Hope that helps...

**Comment by gabaozin** [comment]
Hi, thanks for such a direct and clear message.

You are right about the fact that some versions do work from the app.

I feel it's a shame to not have the 12c specifically as many people from finance areas and related consider it to be a classic one. 

Not sure if there are even better versions for the same purpose, but now I can test them out.

Your procedure works perfectly.

Although for those just looking to have access to the HP12c like me, I just discovered a great resource online, which if you allow me to post here, it's the URL below, which you can create a shortcut as a window, serving well as a webapp:

https://stendec.io/ctb/hp12c.html

It is considerably bigger in the screen, offering a more comfortable usage.

Thanks again!

**Comment by macmpi** [comment]
> serving well as a webapp:

... as long as you have an internet connection, and do not mind sending-over personal data to some ads services.
`<script async src="https://pagead2.googlesyndication.com/pagead/js/adsbygoogle.js">`

Application here is as close-as-possible to original pocket devices & no-strings-attached :wink:

**Comment by mike632t** [comment]
> I feel it's a shame to not have the 12c specifically as many people from finance areas and related consider it to be a classic one

Of course the 12C is included, as well as several other financial models like the HP70, HP80, HP22, HP37E, and HP38E/C so you are spoilt for choice!

The 12c ROM is [here](https://raw.githubusercontent.com/brouhaha/nonpareil/main/ncd/12c/12c.rom), so all you need to get the HP12C emulator working are the following commands:
```
wget https://raw.githubusercontent.com/brouhaha/nonpareil/main/ncd/12c/12c.rom
mv 12c.rom $HOME/.var/app/io.github.mike632t.x11-calc/data/x11-calc/x11-calc-12c.rom
```
If you want to make the window a bit bigger you just need to download and compile the source yourself as you can increase the window size using SCALE_HEIGHT and SCALE_WIDTH.  

![image](https://github.com/mike632t/x11-calc/assets/1799044/bb7c70ef-4066-4739-b05c-dabea6e733d2)

If you think the ability to increase the size of the window by 25% at runtime would be useful could you open a separate issue and I'll see what I can come up with.  Continuous zooming isn't an option currently (X11 does not have scale-able fonts) but adding a command line option to make the window bigger should not be too difficult.

**Comment by gabaozin** [comment]
Very good points from both of you.

Yes, I would really like that. Ok, I'll do that.

Thanks guys!!

**Comment by mike632t** [comment]
Missing firmware is now included in v0.14.0147 - FlatHub has been updated.

Please test and close issue - Thank you
